### PR TITLE
[CELEBORN-629][DOC] Add  doc about enable rac-awareness

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?

Add doc about enabling rac-awareness

### Why are the changes needed?

Document new features.

### Does this PR introduce _any_ user-facing change?

Yes, the docs are updated.

### How was this patch tested?

截屏2023-06-02 下午3 19 10

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @pan3793 @waitinfuture

File: docs/configuration/index.md [hunk 1]
```diff
@@ -151,6 +151,35 @@ If you have 8192 mapper tasks, you could set `spark.celeborn.push.maxReqsInFligh
 
 If `celeborn.worker.flush.buffer.size` is 256 KB, we can have total slots up to 327680 slots.
 
+## Rack Awareness
+
+Celeborn can be rack-aware by set conf `celeborn.client.reserveSlots.rackware.enabled` to `true` on client side.
```
Comment (pan3793) [inline docs/configuration/index.md:156]: ... by setting ...

File: docs/configuration/index.md [hunk 2]
```diff
@@ -151,6 +151,35 @@ If you have 8192 mapper tasks, you could set `spark.celeborn.push.maxReqsInFligh
...
+Shuffle partition block replica placement will use rack awareness for fault tolerance by placing one shuffle partition replica
+on a different rack. This provides data availability in the event of a network switch failure or partition within the cluster.
+
+Celeborn master daemons obtain the rack id of the cluster workers by invoking either an external script or java class as specified by configuration files.
+Using either the java class or external script for topology, output must adhere to the java `org.apache.hadoop.net.DNSToSwitchMapping` interface.
+The interface expects a one-to-one correspondence to be maintained and the topology information in the format of `/myrack/myhost`,
+where `/` is the topology delimiter, `myrack` is the rack identifier, and `myhost` is the individual host.
+Assuming a single `/24` subnet per rack, one could use the format of `/192.168.100.0/192.168.100.5` as a unique rack-host topology mapping.
+
+To use the java class for topology mapping, the class name is specified by the `celeborn.hadoop.net.topology.node.switch.mapping.impl` parameter in the master configuration file.
```
Comment (pan3793) [inline docs/configuration/index.md:166]: java class => Java class, multi places

Comment (AngersZhuuuu) [reply docs/configuration/index.md]: Done

File: docs/configuration/index.md [hunk 3]
```diff
@@ -151,6 +151,35 @@ If you have 8192 mapper tasks, you could set `spark.celeborn.push.maxReqsInFligh
...
+Using either the Java class or external script for topology, output must adhere to the java `org.apache.hadoop.net.DNSToSwitchMapping` interface.
+The interface expects a one-to-one correspondence to be maintained and the topology information in the format of `/myrack/myhost`,
+where `/` is the topology delimiter, `myrack` is the rack identifier, and `myhost` is the individual host.
+Assuming a single `/24` subnet per rack, one could use the format of `/192.168.100.0/192.168.100.5` as a unique rack-host topology mapping.
+
+To use the Java class for topology mapping, the class name is specified by the `celeborn.hadoop.net.topology.node.switch.mapping.impl` parameter in the master configuration file.
+An example, `NetworkTopology.java`, is included with the Celeborn distribution and can be customized by the Celeborn administrator. 
+Using a Java class instead of an external script has a performance benefit in that Celeborn doesn't need to fork an external process when a new worker node registers itself.
+
+If implementing an external script, it will be specified with the `celeborn.hadoop.net.topology.script.file.name parameter` in the master side configuration files.
```
Comment (pan3793) [inline docs/configuration/index.md:170]: `celeborn.hadoop.net.topology.script.file.name parameter` is it desired or a typo?

**Review by pan3793 — APPROVED** [review]

File: docs/configuration/index.md (hunk 3 above)
Comment (AngersZhuuuu) [reply docs/configuration/index.md:170]: > `celeborn.hadoop.net.topology.script.file.name parameter` is it desired or a typo?

typo

Comment (AngersZhuuuu) [reply docs/configuration/index.md]: Fixed

**Comment by pan3793** [comment]
Thanks, merged to master/0.3

**Pull request closed without merging**